xbmc/music/MusicDatabase.cpp

```diff
@@ -5954,7 +5954,12 @@ bool CMusicDatabase::GetArtistArtForItem(int mediaId, const std::string &mediaTy
     if (NULL == m_pDB.get()) return false;
     if (NULL == m_pDS2.get()) return false; // using dataset 2 as we're likely called in loops on dataset 1
 
-    std::string sql = PrepareSQL("SELECT type,url FROM art WHERE media_id=(SELECT idArtist from %s_artist WHERE id%s=%i AND iOrder=0) AND media_type='artist'", mediaType.c_str(), mediaType.c_str(), mediaId);
+    std::string sql;
+    if (mediaType == MediaTypeAlbum)
+      sql = PrepareSQL("SELECT type, url FROM art WHERE media_id=(SELECT idArtist from album_artist WHERE idAlbum=%i AND iOrder=0) AND media_type='artist'", mediaId);
+    else
+      sql = PrepareSQL("SELECT type, url FROM art WHERE media_id=(SELECT idArtist from song_artist WHERE idSong=%i AND idRole=1 AND iOrder=0) AND media_type='artist'", mediaId);
```

Comment:
Can you break these statements into multiple lines so they are better readable and also explain the idRole=1 with an -- comment. Just say what idRole is.